src/main/java/net/mcreator/workspace/resources/ExternalTexture.java

```diff
@@ -0,0 +1,172 @@
+/*
+ * MCreator (https://mcreator.net/)
+ * Copyright (C) 2012-2020, Pylo
+ * Copyright (C) 2020-2024, Pylo, opensource contributors
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ */
+
+package net.mcreator.workspace.resources;
+
+import net.mcreator.generator.GeneratorConfiguration;
+import net.mcreator.io.zip.ZipIO;
+import net.mcreator.plugin.modapis.ModAPIImplementation;
+import net.mcreator.plugin.modapis.ModAPIManager;
+import net.mcreator.ui.workspace.resources.TextureType;
+import net.mcreator.util.image.EmptyIcon;
+import net.mcreator.workspace.Workspace;
+import org.apache.commons.io.FilenameUtils;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.fife.rsta.ac.java.buildpath.LibraryInfo;
+
+import javax.imageio.ImageIO;
+import javax.swing.*;
+import java.io.File;
+import java.io.IOException;
+import java.util.*;
+import java.util.zip.ZipEntry;
+import java.util.zip.ZipFile;
+
+/**
+ * A texture loaded from vanilla MC asset archives or from mods downloaded by API plugins.
+ */
+public final class ExternalTexture extends Texture {
+
+	private static final Logger LOG = LogManager.getLogger(ExternalTexture.class);
+
+	private static final Map<Workspace, Map<CacheIdentifier, Map<String, Texture>>> CACHE = new HashMap<>();

```

Comment:
There may be a problem since Workspace does not implement hashValue. Not sure how this will behave in such case

Reply:
Would it be better to use workspace folder `File` objects for this purpose in this case?

Reply:
Or implementing hashCode/equals, but may make sense to do this in another PR

Reply:
:heavy_check_mark: Fixed